util/cli/recoverer.go

```diff
@@ -0,0 +1,31 @@
+package cli
+
+import (
+	"context"
+
+	"github.com/src-d/lookout"
+	"github.com/src-d/lookout/util/ctxlog"
+)
+
+// RunWatcher starts the watcher with the given handler callback. On failure
+// the watcher is restarted
+func RunWatcher(ctx context.Context, watcher lookout.Watcher, eventHandler lookout.EventHandler) error {
+	errCh := make(chan error, 1)
+	for {
+		go func() {
+			err := watcher.Watch(ctx, eventHandler)
+			errCh <- err
+		}()
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case err := <-errCh:
+			if err != nil {
+				ctxlog.Get(ctx).Errorf(err, "error during watcher.Watch, restarting watcher")
```

Comment:
let's make sure this log will include an event-id, similar to what we have in https://github.com/src-d/lookout/pull/290/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR42

Reply:
We don't have an event here, and the error may even happen outside of the context of an individual event.
Could you please be more specific how this could be improved?